src/array/at.ts

```diff
@@ -14,12 +14,17 @@
  * console.log(result); // [20, 40, 50]
  */
 export function at<T>(arr: readonly T[], indices: number[]): Array<T | undefined> {
-  const result: Array<T | undefined> = new Array(indices.length);
+  const result = new Array<T | undefined>(indices.length);
+  const length = arr.length;
 
   for (let i = 0; i < indices.length; i++) {
-    const index = indices[i];
+    let index = indices[i];
 
-    result[i] = arr.at(index);
+    if (index < 0) {
+      index += length;
+    }
+
+    result[i] = index >= 0 && index < length && Number.isInteger(index) ? arr[index] : undefined;
```

Comment:
Since the `at` function we're working with isn't the compatibility version, I think we should follow the behavior of the native `Array#at(...)` method. 

Otherwise, it could be confusing, because `at(arr, indices)` would behave differently from `indices.map(index => arr.at(index))`. I do appreciate the performance benefits, though.

What do you think?

Reply:
If you prefer to use the native `.at()` for easier maintenance at the cost of performance, then I'm fine with closing this PR.

However, if you want better performance, then see the updated code. I changed the implementation to match that of `Array.prototype.at` and it's around 3.7x to 12x faster than the current version.

![Benchmark results](https://github.com/user-attachments/assets/9ed0a011-1d97-4954-96b7-9d8d1f565bdc)

---

**Because the current and this implementation are incompatible with lodash's, should we create `at()` in `compat`?**

Reply:
If performance is this different, I guess we might go with the improvement you made. Thanks!

I guess we should create `at()` in `compat` to address what you said.